kubernetes/base/config/kube_config.py

```diff
@@ -860,32 +860,36 @@ def load_kube_config_from_dict(config_dict, context=None,
 def new_client_from_config(
         config_file=None,
         context=None,
-        persist_config=True):
+        persist_config=True,
+        client_configuration=None):
     """
     Loads configuration the same as load_kube_config but returns an ApiClient
     to be used with any API object. This will allow the caller to concurrently
     talk with multiple clusters.
     """
-    client_config = type.__call__(Configuration)
+    if client_configuration is None:
+        client_configuration = type.__call__(Configuration)
     load_kube_config(config_file=config_file, context=context,
-                     client_configuration=client_config,
+                     client_configuration=client_configuration,
                      persist_config=persist_config)
-    return ApiClient(configuration=client_config)
+    return ApiClient(configuration=client_configuration)
 
 
 def new_client_from_config_dict(
         config_dict=None,
         context=None,
         persist_config=True,
-        temp_file_path=None):
+        temp_file_path=None,
+        client_configuration=None):
     """
     Loads configuration the same as load_kube_config_from_dict but returns an ApiClient
     to be used with any API object. This will allow the caller to concurrently
     talk with multiple clusters.
     """
-    client_config = type.__call__(Configuration)
+    if client_configuration is None:
+        client_configuration = type.__call__(Configuration)
     load_kube_config_from_dict(config_dict=config_dict, context=context,
-                               client_configuration=client_config,
+                               client_configuration=client_configuration,
                                persist_config=persist_config,
                                temp_file_path=temp_file_path)
-    return ApiClient(configuration=client_config)
+    return ApiClient(configuration=client_configuration)
```

Comment:
with this PR, how would you set the proxy to resolve https://github.com/kubernetes-client/python/issues/1967? 
```
my_client.configuration.proxy = "MY_HTTP_PROXY"
client.CoreV1Api(api_client=my_client).list_namespace()
```

Reply:
This allowed me to have my client working perfectly behind a proxy:

```python
my_client_configuration = client.Configuration
my_client_configuration.proxy = os.getenv("HTTP_PROXY", None)
k8s_cluster_client = config.new_client_from_config(config_file=my_config_file, client_configuration=my_client_configuration)
```

With this solution, you need to find the env vars yourself.
But in a second time I could improve this to find those env vars automatically.